File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/2.java
public class AFDParDeZeros {
    public static boolean parDeZeros(String s) {
        String estado = "q0";
        for (char c : s.toCharArray()) {
            if (estado.equals("q0")) {
                if (c == '0') estado = "q1";
                else if (c == '1') estado = "q0";
            } else if (estado.equals("q1")) {
                if (c == '0') estado = "q0";
                else if (c == '1') estado = "q1";
            }
        }
        return estado.equals("q0");
    }

    public static void main(String[] args) {
        System.out.println(parDeZeros("101"));  // True
        System.out.println(parDeZeros("100"));  // False
    }
}

File: Computabilidade-e-Complexidade-de-Algoritmos-main/python/2.py
def afd_par_de_zeros(s):
    estado = 'q0'
    for char in s:
        if estado == 'q0':
            if char == '0':
                estado = 'q1'
            elif char == '1':
                estado = 'q0'
        elif estado == 'q1':
            if char == '0':
                estado = 'q0'
            elif char == '1':
                estado = 'q1'
    return estado == 'q0'

# Testes
print(afd_par_de_zeros("101"))  # True
print(afd_par_de_zeros("100"))  # False
